### PR TITLE
Add RPM SPEC file

### PR DESCRIPTION
**Pull request by navidshaikh**
Minimize the file diff between midstream and downstream.

### CODE REVIEW

**Comment by navidshaikh** [comment]
cc: @rhuss @warrenvw

File: RELEASE.md [hunk 1]
```diff
@@ -98,17 +98,21 @@ $ git push
...
-### Update Dockerfile in release branch with correct version:
+### Update Dockerfiles and RPM SPEC file in release branch with correct version:
 ```bash
-# Update Dockerfile with current release number:
+# Update Dockerfiles namely: Dockerfile.rhel and Dockerfile.cliartifacts.rhel with current release number:
 # - Change value of tag to current release (RUN TAG="v0.14.0" make build)
 # - Change value of LABEL 'version' to current release (v0.14.0)
-$ vi Dockerfile
+$ vi Dockerfile.rhel
+$ vi Dockerfile.cliartifacts.rhel
```
Comment (navidshaikh) [inline RELEASE.md]: @warrenvw : FYI - We'd need renaming these Dockerfiles after sync at their respective dist git repos too.

File: openshift-serverless-clients.spec [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Version:        %{kn_version}
+Release:        %{kn_release}%{?dist}
+Summary:        %{product_name} client kn CLI binary
+License:        ASL 2.0
+URL:            https://github.com/openshift/knative-client/tree/release-%{kn_cli_version}
+
+ExclusiveArch:  x86_64
+
+Source0:        %{source_tar}
+BuildRequires:  gcc
```
Comment (rhuss) [inline openshift-serverless-clients.spec]: why ?

File: openshift-serverless-clients.spec [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+install -p -m 755 kn-darwin-amd64 %{buildroot}/%{_datadir}/%{name}-redistributable/macos/kn-darwin-amd64
+install -p -m 755 kn-windows-amd64.exe %{buildroot}/%{_datadir}/%{name}-redistributable/windows/kn-windows-amd64.exe
+
+%files
+%license LICENSE
+%{_bindir}/kn
+
+%package redistributable
+Summary:        %{product_name} client CLI binaries for Linux, macOS and Windows
+BuildRequires:  gcc
```
Comment (rhuss) [inline openshift-serverless-clients.spec]: ?

File: openshift-serverless-clients.spec [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+BuildRequires:  golang >= %{golang_version}
+Provides:       %{package_name}-redistributable
```
Comment (rhuss) [inline openshift-serverless-clients.spec]: what's the diff between the normal package and the `-redistributable` package ?

Comment (navidshaikh) [reply openshift-serverless-clients.spec]: normal package for installing `kn` at RHEL machines.

`-redistributable` RPM to package cross-platform binaries together (line no 59 onwards)

excerpts:
```
%files redistributable
%license LICENSE
%dir %{_datadir}/%{name}-redistributable/linux/
%dir %{_datadir}/%{name}-redistributable/macos/
%dir %{_datadir}/%{name}-redistributable/windows/
%{_datadir}/%{name}-redistributable/linux/kn-linux-amd64
%{_datadir}/%{name}-redistributable/macos/kn-darwin-amd64
%{_datadir}/%{name}-redistributable/windows/kn-windows-amd64.exe
```

later these (brew built) binaries are extracted and used for signing+notarizing and sync at mirror.

**Comment by navidshaikh** [comment]
/hold

@rhuss: Testing a scratch build by removing `gcc` from BuildRequires per your review.

**Comment by navidshaikh** [comment]
@rhuss: `gcc` wasn't required for building, tested a scratch build removing it, works fine! Thanks for the review! need lgtm again ;-)

/hold cancel

File: openshift-serverless-clients.spec (hunk 2 above)
Comment (navidshaikh) [reply openshift-serverless-clients.spec]: removed

File: openshift-serverless-clients.spec (hunk 3 above)
Comment (navidshaikh) [reply openshift-serverless-clients.spec]: removed

**Pull request merged**